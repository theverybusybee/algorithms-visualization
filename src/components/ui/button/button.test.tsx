import renderer from "react-test-renderer";
import { Button } from "./button";

describe("button ui-component", () => {
  it("button component has inner text", () => {
    const buttonWithText = renderer
      .create(<Button text="Подтвердить" />)
      .toJSON();
    expect(buttonWithText).toMatchSnapshot();
  });

  it("button component doesn't have inner text", () => {
    const buttonWithoutText = renderer.create(<Button />).toJSON();
    expect(buttonWithoutText).toMatchSnapshot();
  });

  it("button is disabled", () => {
    const disabledButton = renderer.create(<Button disabled />).toJSON();
    expect(disabledButton).toMatchSnapshot();
  });

  it("button is loading", () => {
    const loadingButton = renderer.create(<Button isLoader={true} />).toJSON();
    expect(loadingButton).toMatchSnapshot();
  });

  // it("button works correctly after click event", () => {
  //   const testFunction = jest.fn();
  //   render(<Button onClick={testFunction} />);
  //   fireEvent.click(screen.getByRole("button"));
  //   expect(testFunction).toHaveBeenCalled();
  // });
});
