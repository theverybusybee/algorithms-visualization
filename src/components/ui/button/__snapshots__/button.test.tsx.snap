// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button ui-component button component doesn't have inner text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="submit-button"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button ui-component button component has inner text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="submit-button"
  type="button"
>
  <p
    className="text undefined"
  >
    Подтвердить
  </p>
</button>
`;

exports[`button ui-component button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="submit-button"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button ui-component button is loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="submit-button"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
